File: test/src/utils/char_lookup.cpp
/**
 * @brief
 *
 * @file
 * @ingroup test
 */

#include "common.hpp"

#include "lib/utils/tools/char_lookup.hpp"

using namespace genesis::utils;

TEST(CharLookup, Simple)
{
    auto cl = CharLookup();

    cl.set_selection( "abc" );
    EXPECT_TRUE(  cl.get('a') );
    EXPECT_FALSE( cl.get('A') );

    cl.set_range( 'G', 'L' );
    EXPECT_FALSE( cl.get('F') );
    EXPECT_TRUE(  cl.get('I') );

    cl.set_char( 'I', false );
    EXPECT_FALSE( cl.get('I') );
}

File: python/giannis/test/src/utils/tools/char_lookup.py
import sys

sys.path.append('./bin')

from yoyo.genesis.utils import *

##################################################
#################### SIMPLE ######################

cl = CharLookup_bool_t()

cl.set_selection( "abc", True );
# SIMPLE-1
if not cl.get('a'):
	print("Error at SIMPLE-1")
# SIMPLE-2
if cl.get('A'):
	print("Error at SIMPLE-2")

cl.set_range( 'G', 'L', True );
# SIMPLE-3
if cl.get('F'):
	print("Error at SIMPLE-3")
# SIMPLE-4
if not cl.get('I'):
	print("Error at SIMPLE-4")

cl.set_char( 'I', False );
# SIMPLE-5
if cl.get('I'):
	print("Error at SIMPLE-5")

##################################################

print("\nDone")
